src/app/plugins/scores.plugin.ts

```diff
@@ -11,299 +12,119 @@ export class ScoresPlugin extends Plugin {
-      }
-
-      visitorData = visitorData[0].split('%20');
-      homeData = homeData[0].split('%20');
-
-      const visitorName = this._getTeamNameFromTeamData(visitorData);
-      const homeName = this._getTeamNameFromTeamData(homeData);
-
-      matchups.push(`${visitorName}@${homeName}`);
-    }
-
-    let numFields = 0;
-    let embed = this._getNewEmbed();
-
-    //If use types 'list phone' then show the phone version
-    //This will technically work if the user types anything after 'list'
-    if (teamArg.split(' ').length == 1) {
-      matchups.forEach((matchup) => {
-        if (numFields == 25) {
-          numFields = 0;
-          embedBuffer.push(embed);
-          embed = this._getNewEmbed();
-        }
-
-        const [visitor, home] = matchup.split('@');
-        embed.addField(visitor, home, true);
-        numFields++;
-      });
+  private async _getGame(url: string, teamName: string): Promise<espn.Event> {
+    const games = await this._getGames(url);
+    const targetGame = games.filter((game) =>
+      game.competitions[0].competitors.some(
+        (competitor: espn.Competitor) =>
+          competitor.team.location.toLowerCase() === teamName ||
+          competitor.team.abbreviation.toLocaleLowerCase() === teamName ||
+          competitor.team.name.toLowerCase() == teamName
+      )
+    );
 
-      //Push last embed
-      embedBuffer.push(embed);
-    } else {
-      embed.setDescription(matchups.join('\n').replace(/@/g, ' at '));
-      message.reply(embed);
-    }
-    embedBuffer.forEach((e) => {
-      message.channel.send(e);
-    });
-  }
-  private _getNewEmbed() {
-    const embed = new RichEmbed();
-    embed.setTitle('Current games available');
-    embed.setColor('#7289da');
-    return embed;
+    return targetGame[0]; //There should theoretically be match
   }
 
-  private _evaluateScores(a: number, b: number): number {
-    if (a == b) return 0; //Tied
-    return a > b ? 1 : -1;
+  private async _getGames(url: string): Promise<espn.Event[]> {
+    const response = (await this.container.httpService.get(url)).data;
+    const responseData: espn.Sample = (response as Object) as espn.Sample;
+    return responseData.events;
   }
 
-  private _sendHelp(message: IMessage) {
-    message.reply({
-      embed: {
-        title: 'Scores Plugin Help',
-        color: '7506394',
-        fields: [
-          { name: 'Supported Sports', value: 'NCAA NFL MLB NBA' },
-          {
-            name: 'Inputting Teams',
-            value:
-              'For NCAA: Use their most common name\nFor Professional Sports: Use their city only\n',
-          },
-          {
-            name: 'Examples',
-            value:
-              '!scores NCAA UCF\n!scores NCAA Florida State\n!scores NFL Seattle\n!scores MLB Atlanta',
-          },
-        ],
-      },
-    });
+  private _getIsVisitor(game: espn.Event, teamName: string): boolean {
+    const visitorTeam: espn.Team = game.competitions[0].competitors[1].team;
```

Comment:
how confident are you that you wont run into an out of bounds error here?

Reply:
I ran a forEach to check the length of these arrays and none of them were 0.
I'll run a check anyways